blackbox/bwc/create_bwc_index.py

```diff
@@ -27,18 +27,25 @@
 
 
 def compress_index(version, data_dir, output_dir):
-    compress(data_dir, output_dir, 'bwc-index-%s.zip' % version, 'data')
+    compress(data_dir, output_dir, 'bwc-index-%s.zip' % version)
 
 
-def compress(data_dir, output_dir, zipfile, directory):
+def compress(data_dir, output_dir, target):
     abs_output_dir = os.path.abspath(output_dir)
-    zipfile = os.path.join(abs_output_dir, zipfile)
-    if os.path.exists(zipfile):
-        os.remove(zipfile)
-    olddir = os.getcwd()
-    os.chdir(os.path.dirname(data_dir))
-    subprocess.check_call('zip -r %s %s' % (zipfile, directory), shell=True)
-    os.chdir(olddir)
+    target = os.path.join(abs_output_dir, target)
+    if os.path.exists(target):
+        os.remove(target)
+    zipf = zipfile.ZipFile(target, 'w', zipfile.ZIP_DEFLATED)
```

Comment:
`zipfile.ZipFile` is an autoclosable, so you can use it with a `with` statement:
```py
with zipfile.ZipFile(target, 'w', zipfile.ZIP_DEFLATED) as zipf:
    zipdir(data_dir, zipf, os.path.dirname(data_dir))
````